### PR TITLE
Add nuget package badge and link

### LINKED ISSUE #148: Add links to preview release NuGet packages
Add links to preview release (v8.0.0-preview*) NuGet packages in the README.md or on the release description page.

### PR DESCRIPTION
**Pull request by WeihanLi**
**Closes #148 **

Preview here: https://github.com/WeihanLi/dotnet-CommunityTookit/blob/nuget-package-enhancement/README.md

image

Add the NuGet package badge and link to the NuGet package, hope it would help developers find the NuGet packages more easily

## PR Checklist

- [x] Created a feature/dev branch in your fork (vs. submitting directly from a commit on main)
- [x] Based off latest main branch of toolkit
- [x] PR doesn't include merge commits (always rebase on top of our main, if needed)
- [ ] Tested code with current [supported SDKs](../#supported)
- [ ] New component
  - [ ] Pull Request has been submitted to the documentation repository [instructions](../blob/main/Contributing.md#docs). Link: 
  - [ ] Added description of major feature to project description for NuGet package (4000 total character limit, so don't push entire description over that)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Header has been added to all new source files (run _build/UpdateHeaders.bat_)
- [x] Contains **NO** breaking changes
- [ ] Every new API (including internal ones) has full XML docs
- [ ] Code follows all style conventions

### CODE REVIEW

**Comment by Nirmal4G** [comment]
A table format should be a better choice instead of in-lining the badges. They also look mis-aligned!

**Comment by WeihanLi** [comment]
Would it looks better likes follows, maybe need a new block for the table:

Package | Latest stable | Latest Preview
---------|---------------|---------------
`CommunityToolkit.Mvvm`| [CommunityToolkit.Mvvm](https://www.nuget.org/packages/CommunityToolkit.Mvvm/) | [CommunityToolkit.Mvvm](https://www.nuget.org/packages/CommunityToolkit.Mvvm/absoluteLatest)
`CommunityToolkit.Diagnostics`| [CommunityToolkit.Diagnostics](https://www.nuget.org/packages/CommunityToolkit.Diagnostics/) | [CommunityToolkit.Diagnostics](https://www.nuget.org/packages/CommunityToolkit.Diagnostics/absoluteLatest)
`CommunityToolkit.HighPerformance`| [CommunityToolkit.HighPerformance](https://www.nuget.org/packages/CommunityToolkit.HighPerformance/) | [CommunityToolkit.HighPerformance](https://www.nuget.org/packages/CommunityToolkit.HighPerformance/absoluteLatest)
`CommunityToolkit.Common`| [CommunityToolkit.Common](https://www.nuget.org/packages/CommunityToolkit.Common/) | [CommunityToolkit.Common](https://www.nuget.org/packages/CommunityToolkit.Common/absoluteLatest)

**Comment by Nirmal4G** [comment]
This looks nice but add the help links to the package name column and add a description column at the end. Like this.


Package | Latest Stable | Latest Preview | Description
---------------|---------------|---------------|---------------
[`CommunityToolkit.Common`](https://docs.microsoft.com/dotnet/api/?term=communitytoolkit.common) | [CommunityToolkit.Common](https://www.nuget.org/packages/CommunityToolkit.Common/) | [CommunityToolkit.Common](https://www.nuget.org/packages/CommunityToolkit.Common/absoluteLatest) | This library has a set of helper APIs, shared with other `CommunityToolkit` libraries.

**Comment by WeihanLi** [comment]
@Nirmal4G  updated with table, looks better, thanks

File: README.md [hunk 1]
```diff
@@ -11,11 +11,13 @@
...
+[`CommunityToolkit.Common`](https://docs.microsoft.com/dotnet/api/?term=communitytoolkit.common)| [![CommunityToolkit.Common](https://img.shields.io/nuget/v/CommunityToolkit.Common)](https://www.nuget.org/packages/CommunityToolkit.Common/) | [![CommunityToolkit.Common](https://img.shields.io/nuget/vpre/CommunityToolkit.Common)](https://www.nuget.org/packages/CommunityToolkit.Common/absoluteLatest) |  a set of helper APIs shared with other CommunityToolkit libraries.
```
Comment (Nirmal4G) [inline README.md:20]:
```suggestion
Package | Latest stable | Latest Preview | Description
---------|---------------|---------------|------------
[`CommunityToolkit.Common`](https://docs.microsoft.com/dotnet/api/?term=communitytoolkit.common) | [![CommunityToolkit.Common](https://img.shields.io/nuget/v/CommunityToolkit.Common)](https://nuget.org/packages/CommunityToolkit.Common/) | [![CommunityToolkit.Common](https://img.shields.io/nuget/vpre/CommunityToolkit.Common)](https://nuget.org/packages/CommunityToolkit.Common/absoluteLatest) | A set of helper APIs shared with other `CommunityToolkit` libraries.
[`CommunityToolkit.Diagnostics`](https://docs.microsoft.com/windows/communitytoolkit/diagnostics/introduction) | [![CommunityToolkit.Diagnostics](https://img.shields.io/nuget/v/CommunityToolkit.Diagnostics)](https://nuget.org/packages/CommunityToolkit.Diagnostics/) | [![CommunityToolkit.Diagnostics](https://img.shields.io/nuget/vpre/CommunityToolkit.Diagnostics)](https://nuget.org/packages/CommunityToolkit.Diagnostics/absoluteLatest) | A set of helper APIs (specifically, [`Guard`](https://docs.microsoft.com/windows/communitytoolkit/developer-tools/guard) and [`ThrowHelper`](https://docs.microsoft.com/windows/communitytoolkit/developer-tools/throwhelper)) that can be used for cleaner, more efficient and less error-prone argument validation and error checking.
[`CommunityToolkit.HighPerformance`](https://docs.microsoft.com/windows/communitytoolkit/high-performance/introduction) | [![CommunityToolkit.HighPerformance](https://img.shields.io/nuget/v/CommunityToolkit.HighPerformance)](https://nuget.org/packages/CommunityToolkit.HighPerformance/) | [![CommunityToolkit.HighPerformance](https://img.shields.io/nuget/vpre/CommunityToolkit.HighPerformance)](https://nuget.org/packages/CommunityToolkit.HighPerformance/absoluteLatest) | A collection of helpers for working in high-performance scenarios. It includes APIs such as [pooled buffer helpers](https://docs.microsoft.com/windows/communitytoolkit/high-performance/memoryowner), a fast [string pool](https://docs.microsoft.com/windows/communitytoolkit/high-performance/stringpool) type, a 2D variant of `Memory<T>` and `Span<T>` ([`Memory2D<T>`](https://docs.microsoft.com/windows/communitytoolkit/high-performance/memory2d) and [`Span2D<T>`](https://docs.microsoft.com/windows/communitytoolkit/high-performance/span2d)) also supporting discontiguous regions, helpers for bit shift operations (such as [`BitHelper`](https://docs.microsoft.com/windows/communitytoolkit/high-performance/span2d), also used in [Paint.NET](https://getpaint.net)), and more.
[`CommunityToolkit.Mvvm` (aka MVVM Toolkit)](https://docs.microsoft.com/windows/communitytoolkit/mvvm/introduction) | [![CommunityToolkit.Mvvm](https://img.shields.io/nuget/v/CommunityToolkit.Mvvm)](https://nuget.org/packages/CommunityToolkit.Mvvm/) | [![CommunityToolkit.Mvvm](https://img.shields.io/nuget/vpre/CommunityToolkit.Mvvm)](https://nuget.org/packages/CommunityToolkit.Mvvm/absoluteLatest) | A fast, modular, platform-agnostic MVVM library, which is the official successor of `MvvmLight`. It's used extensively in the Microsoft Store and other first party apps.
`CommunityToolkit.Mvvm.SourceGenerators` | | | The source generators to augment the MVVM Toolkit (included in the `CommunityToolkit.Mvvm`)
```

**Review by Nirmal4G — CHANGES_REQUESTED** [review]

**Review by Nirmal4G — APPROVED** [review]

**Comment by hawkerm** [comment]
Thoughts on updating the upper-right package link to?

image

I think it's pointing to 7.1.2 as it's pointing to `Microsoft.Toolkit` as the profile for the owner on NuGet and getting confused with the package for the same name?

**Comment by Nirmal4G** [comment]
If we do that, then we should display both `Microsoft.Toolkit.*` and `CommunityToolkit.*` packages until we migrate away from the former packages.

@Sergio0694 and @michael-hawker Thoughts?

**Comment by Sergio0694** [comment]
> "until we migrate away from the former packages"

Not sure what you mean, we've already migrated away. I literally deprecated the `Microsoft.Toolkit.*` packages on NuGet 🙂

**Comment by Nirmal4G** [comment]
I should have mentioned that "we" is "Community"! Sorry about that. But if it's not needed to display older packages, then we can remove it! Just wanted to be sure!! 😇

**Comment by Nirmal4G** [comment]
PR #439 should merge first. It might break this. So, don't forget to rebase on top of that merge.

**Comment by MisinformedDNA** [comment]
So we need the source generator package listed given it's not a package?

**Comment by Sergio0694** [comment]
No, it should not be listed, it was my mistake adding that initially. It's just not a package, it should not be there 🙂

**Comment by WeihanLi** [comment]
Close it for a clean commit PR

**Comment by Nirmal4G** [comment]
You could've just force-pushed!? Anyways it's fine.

**Pull request closed without merging**